src/pages/index.js

```diff
@@ -1,67 +1,263 @@
     userInfo.setUserInfo({
-      name: formData["name"],
-      description: formData["description"],
+      name: data.name,
+      about: data.about,
     });
-    profileEditPopup.close();
+  })
+  .catch((error) => {
+    console.error("Failed to fetch profile data:", error);
+  });
+
+// -------------------------------------------------------------------------------------
+// Editing UserInfo Popup
+// -------------------------------------------------------------------------------------
+const profileEditPopup = new PopupWithForm("#profile-edit-modal", {
+  handleFormSubmit: (formData) => {
+    profileEditPopup.toggleUploadIndicator(true);
+
+    api
+      .editProfileData(formData)
+      .then((data) => {
+        userInfo.setUserInfo({
+          name: data.name,
+          about: data.about,
+        });
+        profileEditPopup.close();
+      })
+      .catch((error) => {
+        console.log("Error updating Profile:", error);
+      })
+      .finally(() => {
+        profileEditPopup.toggleUploadIndicator(false);
+      });
   },
 });
+
 profileEditPopup.setEventListeners();
 
+// -------------------------------------------------------------------------------------
+// User Avatar Instance
+// -------------------------------------------------------------------------------------
+const userAvatar = new UserAvatar({
+  avatarSelector: ".profile__image",
+});
+
+// -------------------------------------------------------------------------------------
+// Inital User Avatar Population
+// -------------------------------------------------------------------------------------
+api
+  .getUserAvatar()
+  .then((data) => {
+    userAvatar.setUserAvatar({
+      avatar: data.avatar,
+    });
+  })
+  .catch((error) => {
+    console.error("Failed to fetch profile Avatar:", error);
+  });
+
+// -------------------------------------------------------------------------------------
+// Editing UserAvatar Popup
+// -------------------------------------------------------------------------------------
+const avatarEditPopup = new PopupWithForm("#avatar-edit-modal", {
+  handleFormSubmit: (formData) => {
+    const avatarURL = formData["Image URL"];
+    avatarEditPopup.toggleUploadIndicator(true);
+
+    api
+      .editUserAvatar({ avatar: avatarURL })
+      .then((data) => {
+        userAvatar.setUserAvatar({
+          avatar: data.avatar,
+        });
+        avatarEditPopup.close();
+      })
+      .catch((error) => {
+        console.log("Error updating Avatar:", error);
+      })
+      .finally(() => {
+        avatarEditPopup.toggleUploadIndicator(false);
+      });
+  },
+});
+
+avatarEditPopup.setEventListeners();
+
+// -------------------------------------------------------------------------------------
+// Card Section Instance
+// -------------------------------------------------------------------------------------
+const cardSection = new Section(
+  {
+    items: [],
+    renderer: (cardData) => {
+      const card = new Card(
```

Comment:
You should make a special function `createCard` for creating a card not to copy-paste the code when you need to add a new card with `append`.

```js

function createCard(item) {
    // here will be the code of creating
  return cardElement.getView();
}

```

So, here will be


```js
    renderer: (cardData) => {
      const card = createCard(cardData);
      cardSection.appendItem(card);
    },

```